src/services/labwhere/client.js

```diff
@@ -89,5 +95,52 @@ const scanBarcodesInLabwhereLocation = async (
   )
   return { success: response.success, errors: response.errors, message: response.data.message }
 }
+/**
+ * Exhausts the volume of libraries if the location barcode matches the destroy location.
+ *
+ * @param {string} locationBarcode - The barcode of the location.
+ * @param {Array<string>} labwareBarcodes - The barcodes of the labware.
+ * @returns {Promise<Object>} - An object containing the success status and the exhausted libraries.
+ */
+const exhaustLibraryVolumeIfDestroyed = async (locationBarcode, labwareBarcodes) => {
+  if (locationBarcode !== destroyLocation) return { success: false }
+  let librariesToDestroy = []
+
+  //Fetch libraries by filter key
+  const fetchAndMergeLibraries = async (barcodes, filterKey) => {
+    const filterOptions = { filter: { [filterKey]: barcodes.join(',') } }
```

Comment:
Did you try included both source identifier and barcode filters at the same time? Not sure if that only returns records matching one or both of the filters.

Reply:
Yes, I tried. but it only returned the ones matching both filters like an AND operator, which won't give the intended results for this case.